File: Jack/Assets/HalloweenRange/_Scripts/C#/QuestObjectLife.cs
using UnityEngine;
using System.Collections;

public class QuestObjectLife : MonoBehaviour 
{

	public bool isBoss = false;
	
	private float Damage = 1;
	private QuestScript questScript;
	private Shooting playerShooter;
	private EnemyScript enemyScript;

	//---------------------------------------------------------------------------------------------------------	
	void Start () 
	{
		
		questScript = GameObject.Find("Quest").GetComponent<QuestScript>();
		enemyScript = GetComponent<EnemyScript>();
		
		if (isBoss) 
		{
			enemyScript.SetLife((int)questScript.BossLife);
			playerShooter = GetComponent<EnemyScript>().player.GetComponentInChildren<Shooting>();
		}
		else
		{
			enemyScript.SetLife((int)questScript.DefenceLife);
		}
		
		
	}

	//---------------------------------------------------------------------------------------------------------	
	void Update () 
	{
		if (isBoss)  
		{
			Damage = playerShooter.weapon[playerShooter.activeWeapon].damage;
			enemyScript.SetLife((int)questScript.BossLife);
		}
		else
			enemyScript.SetLife((int)questScript.DefenceLife);
	}

	//---------------------------------------------------------------------------------------------------------	
	
	void OnCollisionEnter(Collision collision) 
	{
		
		if (collision.gameObject.tag == "bullet" || collision.gameObject.tag == "enemy")
		{
			if (isBoss)
				questScript.BossLife -= Damage;
			else
				questScript.DefenceLife -= Damage;
		}
	}
	//---------------------------------------------------------------------------------------------------------	
	
	void SetDamaging(float damage) 
	{
		Damage = damage;
		
	}
	//---------------------------------------------------------------------------------------------------------	
}

File: Jack/Assets/HalloweenRange/_Scripts/Js/QuestObjectLife.js
//QuestObjectLife
var isBoss: boolean = false;

private var Damage: float = 1;
private var questScript;
private var playerShooter;
private var enemyScript;

//---------------------------------------------------------------------------------------------------------	

function Start () {

 questScript = GameObject.Find("Quest").GetComponent(QuestScript); 
 
 enemyScript = GetComponent(EnemyScript);
    
 if (isBoss) 
  {
   enemyScript.SetLife(questScript.GetBossLife());
   playerShooter = GetComponent(EnemyScript).player.GetComponentInChildren(Shooting);
  }
   else
      enemyScript.SetLife(questScript.GetDefenceLife());
 
  
}

//---------------------------------------------------------------------------------------------------------	

function Update () {
  if (isBoss)  
    {
     Damage = playerShooter.weapon[playerShooter.activeWeapon].damage;
     enemyScript.SetLife(questScript.GetBossLife());
    }
     else
      enemyScript.SetLife(questScript.GetDefenceLife());
}

//---------------------------------------------------------------------------------------------------------	

function OnCollisionEnter(collision : Collision) {
  
   if (collision.gameObject.tag == "bullet" || collision.gameObject.tag == "enemy")
    if (isBoss)
         questScript.SetBossLife( questScript.GetBossLife() - Damage);
      else
         questScript.SetDefenceLife( questScript.GetDefenceLife() - Damage);
   
}  


//---------------------------------------------------------------------------------------------------------	

function SetDamaging(damage : float) {

  Damage = damage;
  
}
//---------------------------------------------------------------------------------------------------------	